File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/JsonModel/Api/Callback.cs
using Amazon.JSII.JsonModel.Api.Request;
using Newtonsoft.Json;
using System;

namespace Amazon.JSII.JsonModel.Api
{
    [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
    public sealed class Callback
    {
        public Callback
        (
            string callbackId,
            string? cookie = null,
            InvokeRequest? invoke = null,
            GetRequest? get = null,
            SetRequest? set = null
        )
        {
            CallbackId = callbackId ?? throw new ArgumentNullException(nameof(callbackId));
            Cookie = cookie;
            Invoke = invoke;
            Get = get;
            Set = set;
        }

        [JsonProperty("cbid")]
        public string CallbackId { get; }

        [JsonProperty("cookie", NullValueHandling = NullValueHandling.Ignore)]
        public string? Cookie { get; }

        [JsonProperty("invoke", NullValueHandling = NullValueHandling.Ignore)]
        public InvokeRequest? Invoke { get; }

        [JsonProperty("get", NullValueHandling = NullValueHandling.Ignore)]
        public GetRequest? Get { get; }

        [JsonProperty("set", NullValueHandling = NullValueHandling.Ignore)]
        public SetRequest? Set { get; }
    }
}

File: packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/api/Callback.java
package software.amazon.jsii.api;

import software.amazon.jsii.Internal;

/**
 * The "callback" struct.
 */
@Internal
public class Callback {
    /**
     * Callback ID.
     */
    private String cbid;

    /**
     * Callback cookie.
     */
    private String cookie;

    /**
     * Invoke request (can be null).
     */
    private InvokeRequest invoke;

    /**
     * Get request (can be null).
     */
    private GetRequest get;

    /**
     * Set request (can be null).
     */
    private SetRequest set;

    /**
     * @return Callback ID.
     */
    public String getCbid() {
        return cbid;
    }

    /**
     * Sets the callback ID.
     * @param cbid The callback ID.
     */
    public void setCbid(final String cbid) {
        this.cbid = cbid;
    }

    /**
      * @return The callback cookie.
      */
    public String getCookie() {
        return cookie;
    }

    /**
     * Sets the callback cookie.
     * @param cookie The cookie
     */
    public void setCookie(final String cookie) {
        this.cookie = cookie;
    }

    /**
     * @return The invoke request.
     */
    public InvokeRequest getInvoke() {
        return invoke;
    }

    /**
     * Sets the invoke request.
     * @param invoke The invoke request.
     */
    public void setInvoke(final InvokeRequest invoke) {
        this.invoke = invoke;
    }

    /**
     * @return The get request.
     */
    public GetRequest getGet() {
        return get;
    }

    /**
     * Sets the "get" request.
     * @param get The "get" request
     */
    public void setGet(final GetRequest get) {
        this.get = get;
    }

    /**
     * @return The "set" request.
     */
    public SetRequest getSet() {
        return set;
    }

    /**
     * Sets the "set" request.
     * @param set The "set" request
     */
    public void setSet(final SetRequest set) {
        this.set = set;
    }
}
